heed/src/env.rs

```diff
@@ -276,8 +276,48 @@ pub enum CompactionOption {
 }
 
 impl Env {
-    /// Returns the size used by all the databases on the disk in the environment without taking the free pages into account.
-    pub fn real_disk_space_size(&self) -> Result<u64> {
+    /// The real size used by this environment on disk.
+    pub fn real_disk_size(&self) -> Result<u64> {
+        let path = if self.contain_flag(Flags::MdbNoSubDir)? {
+            self.path().to_path_buf()
+        } else {
+            self.path().join("data.mdb")
+        };
+
+        Ok(path.metadata()?.len())
+    }
+
+    /// Check if a flag was specified when opening the environment.
+    pub fn contain_flag(&self, flag: Flags) -> Result<bool> {
+        let flags = self.raw_flags()?;
+        println!("{flags:032b} &");
+        println!("{:032b}", flag as u32);
+        let or = flags & (flag as u32);
+        println!("{or:032b}");
+
+        Ok(or != 0)
+    }
+
+    /// Return the raw flags stored the environment was opened with.
+    pub fn raw_flags(&self) -> Result<u32> {
+        let mut flags = std::mem::MaybeUninit::uninit();
+        unsafe { mdb_result(ffi::mdb_env_get_flags(self.env_mut_ptr(), flags.as_mut_ptr()))? };
+        let flags = unsafe { flags.assume_init() };
+
+        Ok(flags)
+    }
+
+    /// The map size that was set when configuring the environment.
+    pub fn map_size(&self) -> Result<usize> {
```

Comment:
I would say that the map size should be returned as an u64 instead, am I right?

Reply:
The ffi returns it as an `usize` so I kept it like that.